### DISCUSSION TITLE
Custom new playlist directory for local disk provider

### DISCUSSION THREAD

**Discussion by Rudd-O** [music-assistant/support#2289] [Feature requests and ideas]
In my current setup, only the Playlists directory under the media folder is writable.  This means I cannot create playlists in the root of the media folder through Music Assistant (which is actually the intended result).

There should be an advanced option for the provider that allows the user to specify a path relative to the media root where all new playlists are created.

Thanks in advance.

**Comment by OzGav** [comment]
All playlists are now stored in the MA database so this should not be required.

**Reply by Rudd-O** [reply]
Thank you.

I understand MA has a playlist provider.

I don't want to store them in the MA database.

I want to continue to store them as M3U on disk using the file system provider.  This is a supported feature today and works very well if MA has write access to the root of the music volume (I tried it earlier today by running MA as root temporarily).

I need this feature because those M3U playlists get synced to various portable music players of mine, and appear in other home file server clients to be used directly there as well.  The M3U playlists I make are [the unit of synchronization](https://github.com/Rudd-O/musictoolbox?tab=readme-ov-file#syncplaylists-the-playlist-synchronizer), so a playlist created through MA and living in its database is useless and non-portable to me.  It's cool that the feature exists, but not useful to those of us who curate playlists of a local music collection.

In my case, MA has write access to the playlists folder and to all the playlists too, which is something I'm willing to do — as opposed to giving MA full write access to the entire music directory, which is *not* something I'm willing to do on a permanent basis (I will not be, and am already no longer, running it as root).

I would also be okay with a feature in the file system provider that lets me specify the full path to the playlist I want to write to, or lets me select from a list of writable playlists.

**Reply by OzGav** [reply]
How about adding a second local provider and moving the playlists

**Reply by Rudd-O** [reply]
I could add a second local provider rooted to `/media/Playlists` indeed but, then, because of the root folder location of that new provider vs. the other provider where the music is actually stored, paths on the playlist could never be made relative to the songs in `/media`.  So that wouldn't work.

Clever idea tho.  I bet there are use cases where that can work.

**Reply by OzGav** [reply]
I see you mentioned running it as root. I have a MA user on my NAS which has read/write only to the music folder and sub folders

**Reply by Rudd-O** [reply]
MA definitely has write access to the playlist folder.  That folder is `ugo+rwx`.

Despite that fact, MA does not let me create playlists on the local file system provider anymore.

**Reply by OzGav** [reply]
I just created a playlist on my NAS tonight and it worked ok. Keep looking at the permission situation.

**Reply by Rudd-O** [reply]
```
ls -la [root@donny Music]# ls -la 
total 120
drwxr-xr-x. 6 rudd-o rudd-o   7 Jun 11 12:12 .
drwxr-xr-x. 7 rudd-o rudd-o   8 Jan  4  2023 ..
-rw-r--r--. 1 rudd-o rudd-o 173 Oct  5  2011 .directory
drwxr-xr-x. 6 rudd-o rudd-o   7 May 17 00:36 Good
drwxrwsr-x+ 2 rudd-o rudd-o 165 Jun  2 10:26 Playlists
drwxr-xr-x. 5 rudd-o rudd-o   5 May 30  2008 Rejected
drwxr-xr-x. 5 rudd-o rudd-o   5 May 17 00:10 Unsorted
[root@donny Music]# ls -la ^C
[root@donny Music]# getfacl Playlists
# file: Playlists
# owner: rudd-o
# group: rudd-o
# flags: -s-
user::rwx
user:music-assistant:rwx
user:nfsnobody:rwx
group::r-x
mask::rwx
other::r-x
default:user::rwx
default:user:music-assistant:rwx
default:user:nfsnobody:rwx
default:group::rwx
default:mask::rwx
default:other::r-x
```

also

```
[root@donny Music]# su - music-assistant -s /bin/bash
[music-assistant@donny ~]$ cd /srv/shared/Entertainment/Music/Playlists/
[music-assistant@donny Playlists]$ touch abc.m3u
[music-assistant@donny Playlists]$ ls -l abc.m3u
-rw-rw-r--+ 1 music-assistant rudd-o 0 Jun 11 12:13 abc.m3u
[music-assistant@donny Playlists]$ 
```

It's not a permission situation at all.

Can you please point me to where the code does the determination of whether playlists may be created or not?  If you do, I can make the code change myself.  Thanks!

**Reply by Rudd-O** [reply]
![image](https://github.com/music-assistant/support/assets/346306/d73c2dad-3724-4726-b442-f860fdd3bcd5)

aha!

**Comment by Rudd-O** [comment]
I'm still working on this!
